droneml_dialog.py

```diff
@@ -1,81 +1,144 @@
 
-        # Add Zoom In Button
-        zoom_in_button = QtWidgets.QPushButton("Zoom In")
-        zoom_in_button.clicked.connect(self.zoom_in)
-        button_layout.addWidget(zoom_in_button)
-
-        # Add Zoom Out Button
-        zoom_out_button = QtWidgets.QPushButton("Zoom Out")
-        zoom_out_button.clicked.connect(self.zoom_out)
-        button_layout.addWidget(zoom_out_button)
-
         # Add Load Raster Button
-        load_raster_button = QtWidgets.QPushButton("Load Raster")
-        load_raster_button.clicked.connect(self.load_raster)
-        load_raster_button.setFixedSize(32, 32)
-        button_layout.addWidget(load_raster_button)
+        run_button = QtWidgets.QPushButton("run")
+        run_button.clicked.connect(self.run_classification)
+        run_button.setFixedSize(64, 32)
+        button_layout.addWidget(run_button)
 
         # Add the button layout to the main layout
-        layout.addLayout(button_layout)
+        self.layout.addLayout(button_layout)
 
         # Set the layout to the dialog
-        self.setLayout(layout)
-
-    def zoom_in(self):
-        self.canvas.zoomIn()
-
-    def zoom_out(self):
-        self.canvas.zoomOut()
-
-    def load_raster(self):
-        # Open a file dialog to select a raster file
-        file_path, _ = QtWidgets.QFileDialog.getOpenFileName(self, "Open Raster", "", "Raster Files (*.tif *.tiff *.img)")
-
-        if file_path:
-            # Load the raster layer
-            raster_layer = QgsRasterLayer(file_path, "Loaded Raster")
-
-            # Check if the raster layer is valid
-            if not raster_layer.isValid():
-                QtWidgets.QMessageBox.critical(self, "Error", "Failed to load the raster file.")
-                return
-
-            # Add the raster layer to the current project
-            QgsProject.instance().addMapLayer(raster_layer)
-
-            # Set the canvas to show the raster layer
-            self.canvas.setLayers([raster_layer])
-
-            # Zoom to the full extent of the raster layer
-            self.canvas.setExtent(raster_layer.extent())
-            self.canvas.refresh()
\ No newline at end of file
+        self.layout.setSpacing(0)
+        self.setLayout(self.layout)
+
+    def run_classification(self):
+        """Run the classification algorithm."""
+
+        # Get current selections
+        raster_layer = QgsProject.instance().mapLayersByName(
+            self.raster_combo.currentText()
+        )[0]
+        vec_positive_layer = QgsProject.instance().mapLayersByName(
+            self.vec_positive_combo.currentText()
+        )[0]
+        vec_negative_layer = QgsProject.instance().mapLayersByName(
+            self.vec_negative_combo.currentText()
+        )[0]
+
+        print(f"Raster Layer: {raster_layer}")
+        print(f"Positive Vector Layer: {vec_positive_layer}")
+        print(f"Negative Vector Layer: {vec_negative_layer}")
+
+        ds_raster = rioxarray.open_rasterio(raster_layer.source())
```

Comment:
non tif-file data source (such as some tiled data providindg service)

Reply:
doced this in issue #7